test/unit/gateways/credorax_test.rb

```diff
@@ -980,6 +980,16 @@ def test_nonfractional_currency_handling
     end.respond_with(successful_authorize_response)
   end
 
+  # Utilizes similar conditional as the Live deadline of August 31
```

Comment:
Would it be possible to add a test that simulates a timestamp after the August 31st deadline to confirm that the new logic switches over? I don't believe that minitest has any methods for doing this easily like rspec does, so it might mean that we need to use dependency injection. Instead of using `Time.now.utc` directly in code, we could make it its own function that can have a value injected:

```
attr_accessor :current_time_test_value

def current_time
  @current_time_test_value || Time.now.utc
end
```

Reply:
Nice - I had an inkling something like this was possible, but didn't quite coalesce. I'll implement this.

Reply:
@Carrigan On consideration, I'm not sure this would actually test anything, since we cannot test the Live case, and the sandbox has already implemented the change. However, simulating a _past_ date for the sandbox would actually be helpful. Does that make sense, or am I missing something?

Reply:
My thinking was that for a unit test, we could set `current_time_test_value` to `August 31, 8:01AM UTC` and then check `assert_match(/a4=200&a1=/, data)`. For the sandbox, I am not sure how valuable it is to test a past date since there will never be a case where we make transactions before 8/22/21 again. Does that make sense?

Reply:
@Carrigan Hm, let me rephrase, sorry if I'm still misunderstanding. The problem is that we cannot simulate the Live switchover date, because `test?` is always true for AM unit tests (it would be nice if we could change this, but I don't know how if it's possible). So we can only test the test/sandbox case. Since the sandbox already switched, the current test is already testing the "after the switch" case.

But adding a test that sets the current time prior to Aug 22 would have parity to the real-current-date for the Live case.

Reply:
Actually I found a way to change the mode from the test. That's probably the best way to go here, along with your suggestion for overwriting the current time.